scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -32,23 +32,43 @@
   ; suppress otherwise identical states so we don't spam.
   (= old-state new-state))
 
-(defn remove-finalization-if-set-and-delete
-  [api-client expected-state-dict pod]
-  (api/remove-finalization-if-set-and-delete api-client pod)
-  expected-state-dict)
+(defn delete-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail. Returns a new expected state
+  dict of nil."
+  [api-client pod]
+  (api/kill-task api-client pod)
+  nil) ; Return new expected state dict of nil.
 
 (defn kill-task
+  "Kill task is the same as deleting a task. I semantically distinguish them. Delete is used for completed tasks that
+  we're done with. Kill is used for possibly running tasks we want to kill so that they fail."
   [api-client expected-state-dict pod]
   (api/kill-task api-client pod)
   expected-state-dict)
 
+(defn weird
```

Comment:
Can we name this `log-weird-state` to make clearer what this function is doing?
Also, why is this a separate function instead of the log being inlined? When inlined, we can also include in the log that the pod is going to be killed.